crates/uv/src/requirements.rs

```diff
@@ -209,6 +211,14 @@ impl RequirementsSpecification {
                 let contents = uv_fs::read_to_string(path)?;
                 let pyproject_toml = toml::from_str::<pyproject_toml::PyProjectToml>(&contents)
                     .with_context(|| format!("Failed to parse `{}`", path.normalized_display()))?;
+                if pyproject_toml.build_system.is_some_and(|build_system| {
+                    build_system
+                        .requires
+                        .iter()
+                        .any(|v| v.name.as_dist_info_name().starts_with("poetry"))
+                }) {
+                    warn_user!("uv does not currently support the Poetry-style pyproject.toml (Your requirements.txt would be empty)");
+                }
```

Comment:
Could we move this down and make it conditional on `requirements` being empty? It's possible that they've added a dependency section we can read even if they're using Poetry.

We have a similar warning for `requirements.txt` files

https://github.com/astral-sh/uv/blob/2ea44d863aca4b13823c027a21755067e2dc0050/crates/requirements-txt/src/lib.rs#L317-L322

Maybe we can phrase it the same with a hint that we can't read Poetry's dependencies?

Reply:
(Is it possible to make this conditional on whether `tool.poetry.dependencies` is populated? I assume not, since we don't include that in TOML schema. If so, this is totally fine.)

Reply:
@zanieb Thank you for your suggestion. I updated my PR and also fixed a test case. Let me know if I miss anything!

@charliermarsh Thank you for your comment. Are you planning to support the poetry-style format? If so, and no one is working on it, I'm interested in contributing.